app/drivers/kscan/kscan_gpio_matrix.c

```diff
@@ -239,6 +275,27 @@ static int kscan_matrix_read(const struct device *dev) {
         k_busy_wait(CONFIG_ZMK_KSCAN_MATRIX_WAIT_BEFORE_INPUTS);
 #endif
 
+#ifdef CONFIG_ZMK_KSCAN_GPIO_PORT
+        for (int p = 0; p < config->ports.len; p++)
+        {
+            const struct device *in_port = config->ports.ports[p].port;
+            gpio_port_get(in_port, &config->ports.ports[p].port_input_state);
```

Comment:
The config object really shouldn't be modified. `struct kscan_matrix_data` is for storing current state data.

It looks like you could make this input state bitfield just be a local variable inside this loop?

Reply:
yes this was not my brightest moment, already done that.